docs/src/pages/customization/typography/CustomResponsiveFontSizes.tsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { createMuiTheme, ThemeProvider } from '@material-ui/core/styles';
+import Typography from '@material-ui/core/Typography';
+
+const theme = createMuiTheme();
+
+theme.typography.h3 = {
```

Comment:
A side note, I was thinking of allowing a plugin like API for the theme:

```jsx
import { createMuiTheme } from '@material-ui/core/styles';

// Create a theme instance.
const theme = createMuiTheme(
  prevTheme => ({
    typography: {
      h3: {
        fontSize: '1.2rem',
        '@media (min-width:600px)': {
          fontSize: '1.5rem',
        },
        [prevTheme.breakpoints.up('md')]: {
          fontSize: '2rem',
        },
      },
    },
  }),
  prevTheme => ({
    overrides: {
      MuiCard: {
        root: prevTheme.typography.h3,
      },
    },
  }),
);

export default theme;
```

What do you think about it?

Reply:
One other pros would be to make this validation logic to always run https://github.com/mui-org/material-ui/blob/020b7469d9ef1b9b545a23a40565a2a9c2efc1fc/packages/material-ui/src/styles/createMuiTheme.js#L46.

Reply:
Currently occupied with fixing and improving performance for the current solution. Please open a new issue with a rationale for a new API.

Reply:
Yeap sorry for the distraction :).